File: app/src/main/java/com/prongbang/localizedapp/views/SettingActivity.java
package com.prongbang.localizedapp.views;

import android.content.res.Configuration;
import android.os.Bundle;
import android.os.Handler;
import android.view.View;

import com.prongbang.localization.base.BaseAppCompatActivity;
import com.prongbang.localization.utils.LocaleHelper;
import com.prongbang.localizedapp.R;

public class SettingActivity extends BaseAppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_setting);

        findViewById(R.id.tvThai).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                LocaleHelper.sendBroadcast(getApplicationContext(), LocaleHelper.changeLocale(getApplicationContext(), LocaleHelper.THAI));
                finishDelay(1000);
            }
        });

        findViewById(R.id.tvEnglish).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                LocaleHelper.sendBroadcast(getApplicationContext(), LocaleHelper.changeLocale(getApplicationContext(), LocaleHelper.ENGLISH));
                finishDelay(1000);
            }
        });
    }

    private void finishDelay(long delayMillis) {
        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                finish();
            }
        }, delayMillis);
    }

    @Override
    public void onConfigurationChanged(Configuration newConfig) {
        super.onConfigurationChanged(newConfig);

        recreate();
    }
}

File: app/src/main/java/com/prongbang/localizedapp/SettingActivity.kt
package com.prongbang.localizedapp

import android.content.Context
import android.content.Intent
import android.content.res.Configuration
import android.os.Bundle
import com.prongbang.localization.ENGLISH
import com.prongbang.localization.LocalizationAppCompatActivity
import com.prongbang.localization.Localize
import com.prongbang.localization.THAI
import com.prongbang.localizedapp.databinding.ActivitySettingBinding

class SettingActivity : LocalizationAppCompatActivity() {

	private val binding: ActivitySettingBinding by lazy {
		ActivitySettingBinding.inflate(layoutInflater)
	}

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setContentView(binding.root)

		binding.apply {
			tvThai.setOnClickListener { setLocale(Localize.THAI) }
			tvEnglish.setOnClickListener { setLocale(Localize.ENGLISH) }
		}
	}

	override fun onConfigurationChanged(newConfig: Configuration) {
		openPrepareLocalize() // used only in setting activity
		super.onConfigurationChanged(newConfig)
	}

	companion object {
		fun navigate(context: Context) {
			context.startActivity(Intent(context, SettingActivity::class.java))
		}
	}
}
